cub/test/catch2_test_device_for_each_in_extents.cu

```diff
@@ -55,6 +55,7 @@ static auto fill_linear_impl(c2h::host_vector<T>& vector, const ExtentType&, siz
   _LIBCUDACXX_TRAILING_REQUIRES(void)((Rank == ExtentType::rank()))
```

Comment:
I believe this should simply not be a trailing return type.

We only really need that for friend functions because there nvcc is broken